src/main/java/io/jenkins/plugins/sprp/YAML_FlowDefinition.java

```diff
@@ -61,8 +74,50 @@ public FlowExecution create(FlowExecutionOwner owner, TaskListener listener,
             throw new IllegalStateException("inappropriate context");
         }
 
-        YamlToPipeline y = new YamlToPipeline();
-        String script = y.generatePipeline(this.scriptPath, listener);
+        WorkflowRun build = (WorkflowRun) exec;
+        WorkflowJob job = build.getParent();
+        BranchJobProperty property = job.getProperty(BranchJobProperty.class);
+
+        Branch branch = property.getBranch();
+        ItemGroup<?> parent = job.getParent();
+
+        if (!(parent instanceof WorkflowMultiBranchProject)) {
+            throw new IllegalStateException("inappropriate context");
+        }
+
+        SCMSource scmSource = ((WorkflowMultiBranchProject) parent).getSCMSource(branch.getSourceId());
+
+        if (scmSource == null) {
+            throw new IllegalStateException(branch.getSourceId() + " not found");
+        }
+        SCMHead head = branch.getHead();
+        SCMRevision tip = scmSource.fetch(head, listener);
+        SCMRevision rev = scmSource.getTrustedRevision(tip, listener);
+        GitSCM gitSCM = (GitSCM) scmSource.build(head, rev);
+
+        this.gitConfig = new GitConfig();
+        this.gitConfig.setGitBranch(property.getBranch().getName());
+
+        if(gitConfig.getGitBranch().contains("PR")){
+            String[] refSpecs = gitSCM.getUserRemoteConfigs().get(0).getRefspec().split("/", 0);
+            gitConfig.setGitBranch(refSpecs[refSpecs.length - 1]);
+        }
+
+        this.gitConfig.setGitUrl(gitSCM.getUserRemoteConfigs().get(0).getUrl());
+        this.gitConfig.setCredentialsId(gitSCM.getUserRemoteConfigs().get(0).getCredentialsId());
+
+        String script;
+        try (SCMFileSystem fs = SCMFileSystem.of(scmSource, head, rev)) {
+            if (fs != null) {
+                InputStream yamlInputStream = fs.child(scriptPath).content();
```

Comment:
please use try-with-resources so that the stream gets closed afterwards

Reply:
This is the only way to for try-with-resource right?
[Documentation](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

Reply:
I think that's right?  Any other comments here?